pkg/operator/Makefile

```diff
@@ -1,121 +1,120 @@
-
-##@ Build
-
-build: generate fmt vet ## Build manager binary.
-	go build -o bin/manager main.go
-
-run: manifests generate fmt vet ## Run a controller from your host.
-	go run ./main.go
-
-docker-build: test ## Build docker image with the manager.
-	docker build -t ${IMG} .
-
-docker-push: ## Push docker image with the manager.
-	docker push ${IMG}
-
-create-install-files:
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
-## Create install files for thundernetes installed without API security
-	$(KUSTOMIZE) build config/default | API_SERVICE_SECURITY=none envsubst > ../${INSTALL_FILES_FOLDER}/operator.yaml
-## Create install files for thundernetes installed with API security	
-	$(KUSTOMIZE) build config/default | API_SERVICE_SECURITY=usetls envsubst > ../${INSTALL_FILES_FOLDER}/operator_with_security.yaml
-## Add prometheus service monitoring & roles to the kustomize output
-	cp ../${INSTALL_FILES_FOLDER}/operator.yaml ../${INSTALL_FILES_FOLDER}/operator_with_monitoring.yaml
-	echo "---" >> ../${INSTALL_FILES_FOLDER}/operator_with_monitoring.yaml
-	$(KUSTOMIZE) build config/prometheus >> ../${INSTALL_FILES_FOLDER}/operator_with_monitoring.yaml
-	cp ../${INSTALL_FILES_FOLDER}/operator_with_security.yaml ../${INSTALL_FILES_FOLDER}/operator_with_security_and_monitoring.yaml
-
-##@ Deployment
-
-install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/crd | kubectl apply -f -
-
-uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/crd | kubectl delete -f -
-
-deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
-	$(KUSTOMIZE) build config/default | envsubst | kubectl apply -f -
-
-undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/default | kubectl delete -f -
-
-CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen: ## Download controller-gen locally if necessary.
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
-
-KUSTOMIZE = $(shell pwd)/bin/kustomize
-kustomize: ## Download kustomize locally if necessary.
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
-
-# go-get-tool will 'go get' any package $2 and install it to $1.
-PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
-define go-get-tool
-@[ -f $(1) ] || { \
-set -e ;\
-TMP_DIR=$$(mktemp -d) ;\
-cd $$TMP_DIR ;\
-go mod init tmp ;\
-echo "Downloading $(2)" ;\
-GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
-rm -rf $$TMP_DIR ;\
-}
+# Image URL to use all building/pushing image targets
```

Comment:
cd pkg/operator && make does not work anymore. 
```
dgkanatsios@Dimitrioss-MacBook-Air operator % make
/Users/dgkanatsios/Desktop/projects/thundernetes/pkg/operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...  
go build -o bin/manager main.go
no required module provides package main.go; to add it:
        go get main.go
make: *** [build] Error 1
```
From the description, it seems that Makefile generated from kubebuilder wanted `main.go` to be in the same folder as the rest of the files maybe?

Reply:
Yes, it does still, unfortunately. 

https://github.com/kubernetes-sigs/kubebuilder/issues/932

So I'll refactor this and move the main.go file back into this path.

Reply:
kubebuilder CLI is happy once more